src/NLSModels.jl

```diff
@@ -340,60 +336,16 @@ function obj(nls :: AbstractNLSModel, x :: AbstractVector)
   return dot(Fx, Fx) / 2
 end
 
-function grad(nls :: AbstractNLSModel, x :: AbstractVector)
-  g = zeros(eltype(x), nls_meta(nls).nvar)
-  return grad!(nls, x, g)
-end
-
 function grad!(nls :: AbstractNLSModel, x :: AbstractVector, g :: AbstractVector)
   increment!(nls, :neval_grad)
   Fx = residual(nls, x)
   return jtprod_residual!(nls, x, Fx, g)
 end
 
-function objgrad(nls :: AbstractNLSModel, x :: AbstractVector)
-  g = zeros(eltype(x), nls_meta(nls).nvar)
-  return objgrad!(nls, x, g)
-end
-
 function objgrad!(nls :: AbstractNLSModel, x :: AbstractVector, g :: AbstractVector)
   increment!(nls, :neval_obj)
   increment!(nls, :neval_grad)
   Fx = residual(nls, x)
   jtprod_residual!(nls, x, Fx, g)
   return dot(Fx, Fx) / 2, g
 end
-
```

Comment:
Hessian functions for a general NLS with constraints cannot be defined, so I removed these and passed the responsibility to specific NLS models.

Reply:
Wouldn't it be sufficient to also define `hess` based on `hess_coord` and ask the user to implement `hess_coord!`?

Reply:
We'd need `nnzj` defined to populate `rows` and `cols` and that is model dependent also.

Reply:
We can figure out the sparsity pattern of J'J, and therefore that of H+J'J.

Reply:
Do you mean we assume that `rows` and `cols` will be well-defined? For instance, in `LLSModel` I assumed that computing the coordinate Hessian would be too expensive and used `nnzh = 0`. Instead, we should compute `nnzh` in the constructor of `LLSModel` using the same computations made here?

Going back to my original statement, what I meant was that if `y` is given, we don't know what to do, because we don't have a way to compute the Hessian of the constraints.